### DISCUSSION TITLE
How to list USB devices?

### DISCUSSION THREAD

**Discussion by jetersen** [siderolabs/talos#5609] [Q&A] [answered]
Is it possible to find the USB /dev list?

**Comment by jetersen** [comment, marked as answer]
Well easier than I thought it would be `talosctl list /dev --long --type l`, need `--type l` to see the symbolic link to understand which device is which.

**Reply by jetersen** [reply]
Sadly it seems that Talos OS is not picking up USB bluetooth dongle

**Reply by frezbo** [reply]
we build talos with the least kernel config, it's possible that the drivers for the usb bluetooth adapter is not enabled. Ref: https://github.com/siderolabs/pkgs/blob/master/kernel/build/config-amd64

**Reply by jetersen** [reply]
@frezbo Reason I was testing it out I was toying with Talos OS as my homelab OS with Home Assistant running on k8s.

Sadly this would be a blocker.

in the config you linked to I am not seeing any mention of the `BT` options

https://www.kernelconfig.io/config_bt

Would it be something you would consider enabling?
Is it even possible to use a custom kernel in Talos?

**Reply by frezbo** [reply]
seems bluetooth support is not enabled in the talos kernel. I think it's fine to enable it for k8s use cases. Could you create an issue with the details and we can discuss as a team. Meanwhile you could use a custom kernel in talos by following this docs: https://www.talos.dev/v1.0/advanced/customizing-the-kernel/

**Reply by jetersen** [reply]
@frezbo created https://github.com/siderolabs/pkgs/issues/486

**Reply by davralin** [reply]
@jetersen you haven't tested with any other usb-devices related to HA also?

I haven't debugged it much yet, but I'm having some issues getting my zigbee-device mounted/found..

**Reply by jetersen** [reply]
@davralin I decided to go for another OS for now.

I did notice it was able to pick up my Sonoff Zigbee USB whether it was usable is another story 😅
